src/main/scala/com/evolutiongaming/bootcamp/streaming/fs2.worksheet.sc

```diff
@@ -91,23 +90,24 @@ val pureCount: Option[Map[String, Int]] = pureLines
   .compile
   .last
 
-// Other F[_]'s can work; .compile requires at least Sync
-// E.g. monix Task can work well
+// Other F[_]'s can work
+// Let's try ZIO
 {
-  import monix.eval.Task
-  import monix.execution.Scheduler
+  import zio.stream.interop.fs2z._
+  import zio.{Runtime, Task, UIO, ZIO, Unsafe}
 
-  val monixLines: Stream[Task, String]       = Stream.eval(Task(inputStr))
-  val monixWordCount: Task[Map[String, Int]] = monixLines
+  val zioLines: Stream[UIO, String]                = Stream.eval(ZIO.succeed(inputStr))
+  val zioWordCount: Task[Option[Map[String, Int]]] = zioLines
     .through(fs2.text.lines)
     .through(toWords)
     .map(_.toLowerCase())
     .scanMap(word => Map(word -> 1))
-    .compile
-    .lastOrError
+    .toZStream()
+    .runLast
 
-  implicit val scheduler: Scheduler = Scheduler(ExecutionContext.parasitic)
-  monixWordCount.runSyncUnsafe()
+  Unsafe.unsafe { implicit unsafe =>
```

Comment:
Are you using `Unsafe` to show something on purpose?

It seems you can just inherit an object from `ZioAppDefault` and you have a `run` method to run any `ZIO`

Reply:
You see it's injected in a worksheet, didn't want to extract it to separate app :)